packages/uniforms/src/createAutoField.tsx

```diff
@@ -46,3 +46,7 @@ export function createAutoField(defaultComponentDetector: ComponentDetector) {
     defaultComponentDetector,
   });
 }
+
+function isConnectedLeafComponent(x: Component): x is ConnectedField<any> {
+  return typeof x === 'object' && 'options' in x && x.options?.kind === 'leaf';
```

Comment:
Isn't the optional chaining redundant here since you check for the existence of `options` property beforehands?
```suggestion
  return typeof x === 'object' && 'options' in x && x.options.kind === 'leaf';
```

Reply:
I just wanted to make sure it'll be fine if someone uses `options: undefined`, because before TypeScript 4.4 it's actually a valid value ([link](https://devblogs.microsoft.com/typescript/announcing-typescript-4-4/#exact-optional-property-types)).